lib/components.js

```diff
@@ -10,7 +10,7 @@ import {
   getTypeFromUnion,
   getComponentOptions
 } from './util';
-
+import { util } from "@rhumbix/tcomb-json-schema"
```

Comment:
I don't see `@rhumbix/tcomb-json-schema` in the `package.json`, so I imagine we would need that in order to import that here, right? I made that change to util.Num in tcomb-json-schema bc of the issue with numbers being strings and failing validation, but I am wondering if we are fixing the wrong issue here and should fix the root issue that Number fields are returning Strings instead of Numbers.

Reply:
Same. Shouldn't this be forcing the values rather than just testing the type? And why the need to also check against `util.Int`? Doesn't `util.Number` handle that? This needs tests as well to ensure it works and see how it handles empty strings.

Reply:
Testing the type properly there will return the right format and parse functions for number. [here](https://github.com/Rhumbix/tcomb-form-native/blob/master/lib/components.js#L288). The `format` function will handle null value and format number into string to show it on textbox UI properly. When save, the `parse` function with parse string back to number to store. So `number` should always be stored as numbers instead of string.

Reply:
I will add some tests for this on RN repo. It's easier to add test there.

Reply:
If it is possible to get the test in this repo that would be best since this is where the functionality lives, and we shouldn't be writing tests for outside libraries in our RN/WA repos. It does appear there is a tests folder in this repo, so might be able to derive from there what you would need to do?

Reply:
hmm, yea, this repo has unit tests using Tape. I haven't used Tape before. But i will add couple tests for this.